### DISCUSSION TITLE
Search filter state is cleared after selection

### DISCUSSION THREAD

**Discussion by ttomaszewski** [dowjones/react-dropdown-tree-select#507] [General]
**Describe the bug**
We are using this plugin (hierarchical mode) for a searchable dropdown which allows multiple selections. I'm seeing a bug where my search filter is cleared after selection.

Is there a way to preserve the search state after selection?

**To Reproduce**
1. Search for an available option
2. Select an option

Observed: Search state is cleared while the search term remains in the input

**Expected behavior**
Search filter is not cleared after a selection

**Screenshots**
https://user-images.githubusercontent.com/37166713/126522878-c740deb5-f002-4c9b-9160-4ddeb6c4b060.mov

**Comment by ldeveber** [comment]
In our particular case, we are updating the list of nodes with the checked state, and passing it back into the plugin.

When this list of nodes changes, either it should continue to be filtered on the search term, or clear the search term when the list of nodes changes?

**Comment by mrchief** [comment]
Thanks for attaching the video. From the video, I can tell that the component is being re-rendered and @ldeveber just confirms my theory. 

The dropdown, once initialized should not be updated with new data unless the data has changed. It'll keep track of selected/deselected nodes so there is no need to re-assign checked nodes.
